dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/queue/MasterPriorityQueueTest.java

```diff
@@ -0,0 +1,92 @@
+package org.apache.dolphinscheduler.service.queue;
+
+import org.apache.dolphinscheduler.common.model.Server;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions;
+
+public class MasterPriorityQueueTest {
+
+
+    @Test
+    public void getOrderedCollection () throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        MasterPriorityQueue queue = new MasterPriorityQueue();
+        Method getOrderedElementsMethod = MasterPriorityQueue.class.getDeclaredMethod("getOrderedElements");
+        getOrderedElementsMethod.setAccessible(true);
+
+        // Test empty queue
+        Server[] emptyElements = (Server[]) getOrderedElementsMethod.invoke(queue);
+        Assertions.assertArrayEquals(emptyElements, new Server[]{});
+
+        // Test queue with fabricated servers
+        queue.putAll(getServerList());
+        Server[] orderElements = (Server[]) getOrderedElementsMethod.invoke(queue);
+        Assertions.assertEquals(extractServerIds(orderElements), Arrays.asList(4, 2, 1, 3));
+
+    }
```

Comment:
Don't use reflection to test private methods

Reply:
Got it, modified.